calabash-cucumber/lib/calabash-cucumber/map.rb

```diff
@@ -143,6 +144,40 @@ def self.assert_map_results(map_results, msg)
         end
       end
 
+      # Evaluating whether a query contain correct predicate selector
+      # returns true if query include correct predicate selector
+      def self.correct_predicate?(query)
+        if !query.match(/{.*}/).nil?
+          str = query.match(/{.*}/)[0]
+          correct = false
+          predicate = %w(BEGINSWITH CONTAINS ENDSWITH LIKE MATCHES)
+          predicate.each do |value|
+            if str.include?(value)
+              correct = true
+              return correct
+            end
+          end
+          if !correct
+            fail "Incorrect predicate used, valid selectors are: #{predicate}"
```

Comment:
Don't use `fail` here.  Raise a RuntimeError.  We don't need to take a screenshot.